test/src/String/test_comparisonFunc.cpp

```diff
@@ -10,32 +10,42 @@
 
 #include <String.h>
 
+#include "StringPrinter.h"
+
 /**************************************************************************************
  * TEST CODE
  **************************************************************************************/
 
 TEST_CASE ("Testing String::equals(const String &) with exit status PASS", "[String-equals-01]")
 {
   arduino::String str1("Hello"), str2("Hello");
-  REQUIRE(str1.equals(str2) == 1);
+  CHECK(str1.equals(str2) == 1);
```

Comment:
Good morning @matthijskooijman 👋 ☕ Sorry for missing this additional PR to #97. I'd outright approve and merge it but for this sequenced `CHECK` constructs.
- Why are using `CHECK` instead of `REQUIRE`?
- Why are there 3 checks within one test case? As I've written in my feedback on #97 the usual testing idiom is: 1 Testcase - 1 Assertion. If you have multiple assertions within one testcase you are testing multiple things which is mudding the water. Please split these tests up.

Reply:
I merged these because they essentially test the same thing. More specifically, they test that `equals`, `==` and `!=` (which are all just different ways to call the same code) behave correctly in the same situation. I could split them up, but that would essentially just mean more duplication of code for creating the test situation (which admittedly isn't much code, but still).

So I'd still think these are better as three checks in a single testcase, but if you insist, I'm also fine with splitting them up.

Reply:
I see :eyes: I've not yet made up my mind on how to best approach these situations where one function is basically just a wrapper and directly calls another function. My main intention behind objecting to your test-code-merging was on the basis that I don't want to set a precedent for bad test design. Looking deeper into the issue I do see the reasons for you doing so. Nonetheless I'd very much prefer `REQUIRE` over `CHECK`. If you could please change that I'll be happy to merge this PR.

Reply:
Ok, I changed the PR as you asked.

Note that in addition to one function that wraps another, a second use of CHECK was where you test one situation, and want to check multiple postconditions of this situation. In this case, assigning a string with `NULL` and checking that the string is not still the same as before and is marked as invalid. I've now duplicated this testcase, which leads to a little bit more duplicate code than the other case. Also, this means that the testcase description now not only needs to indicate what situation is being tested, but also what postcondition is being checked. 

See 7deb046a50b72fd91ebe41e61c684fa8580ec0fd (note that this used to be an added `CHECK` line in another commit, but because it really was off-topic for that commit and now adds 2 testcases instead of just one `CHECK` line, I moved this into a commit of its own).

Reply:
Heh, seems you merged directly after I pushed, while I was making one more tweak to my commits (making the history a little cleaner and adding one more testcase). Now that it's merged, not really relevant to spend more time on.

For anyone reading my previous comment, note that the commit referenced there is thus not actually merged, but is still part of a675d5a5cbb7fc6334f2f873819e582ae6437fb3 (which is merged).

Reply:
Sorry about that :blush:

Reply:
@matthijskooijman Could you maybe open a separate PR cleaning up the test code?

Reply:
It was mostly a cleanup of git history, which I can't fix anymore (but it wasn't a big cleanup) and one added testcase just for consistency, which isn't really important anyway (not worth setting up another PR for, IMHO), so I would just leave this as it is now.

Reply:
Roger that. I'll be sure to wait a bit longer next time :+1: